library/std/src/sys/pal/unix/thread.rs

```diff
@@ -440,13 +440,24 @@ pub fn available_parallelism() -> io::Result<NonZero<usize>> {
                     }
                 }
             }
+
+            let mut ulimit = u64::MAX;
+            #[cfg(any(target_os = "android", target_os = "linux"))]
+            {
+                let mut r: libc::rlimit = unsafe { mem::zeroed() };
+                unsafe {
+                    if libc::getrlimit(libc::RLIMIT_NPROC, &mut r) == 0 {
+                        ulimit = r.rlim_max
```

Comment:
Note, the "soft" `rlim_cur` is the value that is currently enforced, while the "hard" `rlim_max` is a limit on how high an unprivileged process can adjust its own `rlim_cur`.

Reply:
ok